### PR TITLE
feat: migrate app tools deploy plugin and analyze plugin

### PR DESCRIPTION
**Pull request by caohuilin**
## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/toolkit/plugin-v2/src/types/cli/context.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import type { CLIPluginAPI } from './api';
 import type { CLIPlugin, CLIPluginExtends } from './plugin';
 
 export interface Entrypoint {
-  name: string;
+  entryName: string;
```
Comment (zllkjc) [inline packages/toolkit/plugin-v2/src/types/cli/context.ts:11]: 这里改动是为啥，因为类型里 entry 用不到了，只用到 entryName 吗

File: packages/solutions/app-tools/src/plugins/analyze/getHtmlTemplate.ts [hunk 2]
```diff
@@ -43,15 +39,49 @@ const findPartials = (
...
+  };
+  const prepend = (type: keyof HtmlPartials, ...script: string[]) => {
+    script.forEach(item => {
+      partials[type].unshift(item);
+    });
+  };
+  return {
+    top: {
+      append: (...script: string[]) => append(PartialPosition.TOP, ...script),
+      prepend: (...script: string[]) => prepend(PartialPosition.TOP, ...script),
```
Comment (zllkjc) [inline packages/solutions/app-tools/src/plugins/analyze/getHtmlTemplate.ts:58]: So, `append` and `prepend` is the new API for modifyHtmlPartials hook ?

File: packages/solutions/app-tools/src/plugins/deploy/index.ts [hunk 3]
```diff
@@ -21,14 +25,14 @@ const deployPresets: DeployPresetCreators = {
...
-  modernConfig: NormalizedConfig<AppTools>,
+  appContext: AppToolsContext<'shared'>,
+  modernConfig: AppToolsNormalizedConfig,
   deployTarget: DeployTarget,
 ) {
   const { appDirectory, distDirectory } = appContext;
   const { useSSR, useAPI, useWebServer } = getProjectUsage(
     appDirectory,
-    distDirectory,
+    distDirectory!,
```
[the remark this answers is not in the document]
Comment (caohuilin) [reply packages/solutions/app-tools/src/plugins/deploy/index.ts:35]: 因为这个类型是在 plugin-v2 包里面定义的，但是是在 app-tools 里面初始化的

File: packages/toolkit/plugin-v2/src/types/cli/context.ts (hunk 1 above)
Comment (caohuilin) [reply packages/toolkit/plugin-v2/src/types/cli/context.ts:11]: app-tools 里面的 Entrypoint 是用的 entryName，这样可以兼容

File: packages/solutions/app-tools/src/plugins/analyze/getHtmlTemplate.ts (hunk 2 above)
Comment (caohuilin) [reply packages/solutions/app-tools/src/plugins/analyze/getHtmlTemplate.ts:58]: yes

File: packages/toolkit/plugin-v2/src/types/cli/context.ts (hunk 1 above)
Comment (zllkjc) [reply packages/toolkit/plugin-v2/src/types/cli/context.ts:11]: OK

**Review by zllkjc — APPROVED** [review]

**Pull request merged**